### PR TITLE
Add Autosuggest support for the SearchBox

### PR DESCRIPTION
**Pull request by joerideg**

### CODE REVIEW

File: src/search/search-box/search-box.stories.tsx [hunk 1]
```diff
@@ -45,38 +46,38 @@ const searchOptions: Omit<ProductSearchOptions, 'q'> = {
...
   url: 'https://example.com',
-  'facet.version': '3.0'
+  'facet.version': '3.0',
 };
 
 export const Basic: Story = {
   render: (args) => {
     return (
-      <SearchContextProvider>
+      <LimitlessUIProvider>
```
Comment (anandgorantala) [inline src/search/search-box/search-box.stories.tsx:55]: We may have to revisit this usecase, if the searchbox will ever be used this way. The Autosuggest provider could be relevant to wrap the searchbox, but the search page is just one of many pages on a site and it's more likely the consumer will use the hook to do the search and then pass the search response to a results renderer. Let's revisit in a future PR as we add theming to this and start to use this in the code samples

File: src/search/suggestions/suggestions.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    suggestOptions,
+    inputValue: 'chair',
+  };
+
+  it('should render successfully', () => {
+    const {
+      container: { firstElementChild },
+    } = render(<Suggestions {...props} />);
+
+    expect(firstElementChild).toBeTruthy();
```
Comment (anandgorantala) [inline src/search/suggestions/suggestions.spec.tsx:29]: you can change this from `tobeTruthy` to `toMatchSnapshot` to catch any mis-rendering in the future

**Review by anandgorantala — APPROVED** [review]
Left a few minor comments, we can address them in a future PR as we add theming and use this in the code samples

File: src/search/search-box/search-box.stories.tsx (hunk 1 above)
Comment (joerideg) [reply src/search/search-box/search-box.stories.tsx:55]: I'm honestly not sure what it is that you are attempting to indicate. This basic story is just about using the searchbox without autosuggest, which seems to me quite relevant as its a separate feature and a separate API that might or might not be set up for a specific customer right?

File: src/search/suggestions/suggestions.spec.tsx (hunk 2 above)
Comment (joerideg) [reply src/search/suggestions/suggestions.spec.tsx:29]: yea makes sense

File: src/search/search-box/search-box.stories.tsx (hunk 1 above)
Comment (joerideg) [reply src/search/search-box/search-box.stories.tsx:55]: Lets sync over Slack

**Pull request merged**